calico_node/startup/startup.go

```diff
@@ -282,12 +289,15 @@ func parseIPEnvironment(envName, envValue string, version int) *net.IP {
 
 // validateIP checks that the IP address is actually on one of the host
 // interfaces and warns if not.
-func validateIP(ip *net.IP) {
+func validateIP(ipn *net.IPNet) {
 	// No validation required if no IP address is specified.
-	if ip == nil {
+	if ipn == nil {
 		return
 	}
 
+	// Pull out the IP as a net.IP (it has useful string and version methods).
+	ip := net.IP{ipn.IP}
```

Comment:
Alas no.  `ipn` is one of our wrapped IPNet objects which still only contains the golang.net IP and Mask components.

Another reason the wrapping approach is so stinky.